buildSrc/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts

```diff
@@ -0,0 +1,116 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+plugins {
+  `java-library`
+
+  id("net.bytebuddy.byte-buddy")
+
+  id("io.opentelemetry.instrumentation.base")
+  id("io.opentelemetry.instrumentation.javaagent-codegen")
+  id("io.opentelemetry.instrumentation.javaagent-shadowing")
+}
+
+dependencies {
+  // Integration tests may need to define custom instrumentation modules so we include the standard
+  // instrumentation infrastructure for testing too.
+  compileOnly("io.opentelemetry.instrumentation:opentelemetry-instrumentation-api")
+  compileOnly("io.opentelemetry.javaagent:opentelemetry-javaagent-api")
+  compileOnly("io.opentelemetry.javaagent:opentelemetry-javaagent-bootstrap")
+  // Apply common dependencies for instrumentation.
+  compileOnly("io.opentelemetry.javaagent:opentelemetry-javaagent-extension-api") {
+    // OpenTelemetry SDK is not needed for compilation
+    exclude(group = "io.opentelemetry", module = "opentelemetry-sdk")
+    exclude(group = "io.opentelemetry", module = "opentelemetry-sdk-metrics")
+  }
+  compileOnly("io.opentelemetry.javaagent:opentelemetry-javaagent-tooling") {
+    // OpenTelemetry SDK is not needed for compilation
+    exclude(group = "io.opentelemetry", module = "opentelemetry-sdk")
+    exclude(group = "io.opentelemetry", module = "opentelemetry-sdk-metrics")
+  }
+
+  testImplementation("io.opentelemetry.javaagent:opentelemetry-testing-common")
+
+  add("codegen", "io.opentelemetry.javaagent:opentelemetry-javaagent-bootstrap")
+  add("codegen", "io.opentelemetry.javaagent:opentelemetry-javaagent-tooling")
+  add("codegen", "io.opentelemetry.javaagent:opentelemetry-javaagent-extension-api")
+}
+
+val testInstrumentation by configurations.creating {
+  isCanBeConsumed = false
+  isCanBeResolved = true
+}
+
+tasks.named<ShadowJar>("shadowJar").configure {
+  configurations = listOf(project.configurations.runtimeClasspath.get(), testInstrumentation)
+
+  archiveFileName.set("agent-testing.jar")
+}
+
+val agentForTesting by configurations.creating {
+  isCanBeConsumed = false
+  isCanBeResolved = true
+}
+
+dependencies {
+  agentForTesting("io.opentelemetry.javaagent:opentelemetry-agent-for-testing")
+}
+
+// need to run this after evaluate because testSets plugin adds new test tasks
+afterEvaluate {
+  tasks.withType<Test>().configureEach {
+    val shadowJar = tasks.shadowJar.get()
+    val agentShadowJar = agentForTesting.resolve().first()
+
+    inputs.files(agentForTesting)
+    inputs.file(shadowJar.archiveFile)
```

Comment:
I think you can and should depend on the task, not explicitly on its output file.

Reply:
You mean with `dependsOn(shadowJar)` as on line 67? I thought so too, I think @trask added this later finding it wasn't enough. I haven't tried investigating past that but for the purpose of this PR, it's copied over from before any kotlin migrations.